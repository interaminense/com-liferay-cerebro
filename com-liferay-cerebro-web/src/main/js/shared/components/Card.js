import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

class Body extends Component {
	render() {
		const {title} = this.props;

		return (
			<div class="card-body">
				{title && (
					<div class="card-title" key="cardTitle">
						{title}
					</div>
				)}

				{title && <div class="card-divider" key="cardDivider" />}

				{this.props.children}
			</div>
		);
	}
}

Body.PROPS = {
	title: Config.string()
};

class Row extends Component {
	render() {
		return <div class="card-row">{this.props.children}</div>;
	}
}

class Title extends Component {
	render() {
		return <div class="card-title">{this.props.children}</div>;
	}
}

class Card extends Component {
	render() {
		const {children, horizontal} = this.props;

		const classes = getCN('card', {
			horizontal
		});

		return (
			<div {...this.otherProps()} class={classes}>
				{children}
			</div>
		);
	}
}

Card.PROPS = {
	horizontal: Config.bool()
};

Card.Body = Body;
Card.Row = Row;
Card.Title = Title;

export default Card;