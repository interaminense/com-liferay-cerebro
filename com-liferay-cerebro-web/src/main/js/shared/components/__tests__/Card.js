import Card from '../Card';

describe('Card', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Card();

		expect(component).toMatchSnapshot();
	});

	it('should render horizontal', () => {
		component = new Card({
			horizontal: true
		});

		expect(component).toMatchSnapshot();
	});
});