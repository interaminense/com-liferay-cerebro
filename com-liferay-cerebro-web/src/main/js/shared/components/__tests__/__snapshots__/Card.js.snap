// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `<div class="card"></div>`;

exports[`Card should render horizontal 1`] = `<div class="card horizontal"></div>`;
